src/main/java/pl/psnc/dei/service/TranscriptionPlatformService.java

```diff
@@ -209,7 +217,34 @@ public JsonObject fetchTranscriptionUpdate(Transcription transcription) {
 				})
 				.block();
 
-		return JSON.parse(response);
+		final String currentTranscription = retrieveCurrentTranscription(transcription.getTp_id(), response);
+		if (currentTranscription != null) {
+			return JSON.parse(currentTranscription);
+		}
+		return null;
+	}
+
+	private String retrieveCurrentTranscription(String transcriptionId, String enrichments) {
+		JSONArray enrichmentsJson = null;
+		try {
+			JSONParser parser = new JSONParser();
+			enrichmentsJson = (JSONArray) parser.parse(enrichments);
+		} catch (ParseException e) {
+			logger.error("Parsing transcription error", e);
+		}
+
+		if(enrichmentsJson == null) {
+			logger.info("Empty transcription object {}", transcriptionId);
+			return null;
+		}
+
+		for (Object en : enrichmentsJson) {
+			JSONObject json = (JSONObject) en;
+			if (json.getAsString("AnnotationId").equals(transcriptionId)) {
```

Comment:
Actually I thought about the check here - when TP changes the "AnnotationId" field to something else we will have NPE here...